### PR TITLE
Caip222 over universal linking

### PR DESCRIPTION
**Pull request by llbartekll**
# Description

Resolves # (issue)

## Due Dilligence

* [ ] Breaking change
* [ ] Requires a documentation update

### CODE REVIEW

File: Sources/WalletConnectSign/LinkAndRelayDispatchers/SessionRequester.swift [hunk 1]
```diff
@@ -0,0 +1,36 @@
+
+import Foundation
+
+class SessionRequester {
```
Comment (jackpooleywc) [inline Sources/WalletConnectSign/LinkAndRelayDispatchers/SessionRequester.swift:4]:
```suggestion
final class SessionRequester {
```

File: Sources/WalletConnectSign/LinkAndRelayDispatchers/SessionRequestDispatcher.swift [hunk 2]
```diff
@@ -0,0 +1,43 @@
+
+import Foundation
+
+class SessionRequestDispatcher {
```
Comment (jackpooleywc) [inline Sources/WalletConnectSign/LinkAndRelayDispatchers/SessionRequestDispatcher.swift:4]:
```suggestion
final class SessionRequestDispatcher {
```

**Review by jackpooleywc — COMMENTED** [review]
Quick first pass, a few small things, most of them should probably be deferred to follow up PRs. 
Great work on this, couldn't be more excited to see this so close to landing 🚀

File: Tests/WalletConnectKMSTests/EnvelopeTests.swift [hunk 3]
```diff
@@ -8,8 +8,15 @@ final class EnvelopeTests: XCTestCase {
     func testSerialisation() {
         let envelope = Envelope(type: .type1(pubKey: pubKey), sealbox: sealbox)
         let serialised = envelope.serialised()
-        let deserialised = try! Envelope(serialised)
+        let deserialised = try! Envelope(.base64Encoded(serialised))
         XCTAssertEqual(envelope, deserialised)
     }
 
+    func testDeserialise() {
```
[the remark this answers is not in the document]
Comment (llbartekll) [reply Tests/WalletConnectKMSTests/EnvelopeTests.swift:15]: added

File: Sources/WalletConnectKMS/Serialiser/Serializer.swift [hunk 4]
```diff
@@ -6,20 +6,26 @@ public class Serializer: Serializing {
...
+            case .topicNotFound:
+                return "Error: Topic not found."
             }
         }
     }
 
     private let kms: KeyManagementServiceProtocol
     private let codec: Codec
-    private let logger: ConsoleLogging
+    public let logger: ConsoleLogging
```
[the remark this answers is not in the document]
Comment (llbartekll) [reply Sources/WalletConnectKMS/Serialiser/Serializer.swift:28]: so we can turn of/off logging from outside

**Review by jackpooleywc — APPROVED** [review]
:shipit:

**Pull request merged**